### PR TITLE
use strict, strict comaprisons, 200lines

### PR DESCRIPTION
**Pull request by tunnckoCore**
- use `'use strict'`
- use strict comparisons
- minify to 200 lines
- resolve object and resolve array is 99% same, merge them, follow DRY
- use `foreach` for consistent foreach iterator for arrays and objects
- use `object.prototype.toString` instead of just `typeof`
- dont use yoda conditions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -241,10 +201,7 @@ function resolve_array(arr) {
  * @return {String}
  */
 
-function type(v) {
-  return isArray(v)
-    ? 'array'
-    : v.vo
-    ? 'vo'
-    : typeof v;
+function typeOf(v) {
```
Comment (matthewmueller) [inline index.js]: For the purpose of differentiating between arrays, functions and objects, what is the benefit of this over typeof?

File: index.js [hunk 2]
```diff
@@ -241,10 +201,7 @@ function resolve_array(arr) {
...
+  var toString = Object.prototype.toString;
```
Comment (matthewmueller) [inline index.js]: can be moved to the top

File: index.js [hunk 3]
```diff
@@ -1,7 +1,10 @@
+'use strict';
+
 /**
  * Module Dependencies
  */
 
+var foreach = require('foreach');
```
Comment (matthewmueller) [inline index.js]: needs to be added to the package.json

**Comment by matthewmueller** [comment]
looks pretty good! added some comments above.

File: index.js (hunk 3 above)
Comment (tunnckoCore) [inline index.js]: o yea, hah.

File: index.js (hunk 1 above)
Comment (tunnckoCore) [inline index.js]: hm, u have some right. maybe limits to not come instance of class. e.g. `vo(new Fn)` which, yea, im not sure should we handle it or not?

Comment (matthewmueller) [inline index.js]: let's keep it simple and remove this piece for now. if it comes up in an issue we can address it then.

Comment (tunnckoCore) [inline index.js]: k

**Comment by matthewmueller** [comment]
<3 thanks!

**Comment by tunnckoCore** [comment]
cool :dancers:

**Pull request merged**